src/renderer/services/eth2/client/prysm/prysm.ts

```diff
@@ -23,35 +22,27 @@ export class PrysmEth2ApiClient extends AbstractApiClient implements IValidatorB
 
     public constructor(options: IBeaconClientOptions) {
         super();
-        options.urlPrefix = `${options.urlPrefix}${apiPrefix}`;
-        this.url = options.urlPrefix;
+        options.baseUrl = `${options.baseUrl}${apiPrefix}`;
+        this.url = options.baseUrl;
         this.config = options.config;
         this.beacon = new PrysmBeaconApiClient(options);
         this.validator = new PrysmValidatorApiClient(options);
     }
 
-    public static getPrysmBeaconClient(url: string, network: string): PrysmEth2ApiClient | null {
-        const networkConfig = getNetworkConfig(network);
-        if (!networkConfig) {
-            return null;
-        }
-
-        return new PrysmEth2ApiClient({
-            urlPrefix: url,
-            config: networkConfig.eth2Config
-        });
-    }
-
     public async getVersion(): Promise<string> {
         return (await this.beacon.getClientVersion()).toString("ascii");
     }
 
-    public async isSyncing(): Promise<boolean> {
-        const result = await this.beacon.getSyncingStatus();
-        return (typeof result === "boolean") ? result : false;
-    }
-
-    public async getChainHeight(): Promise<string> {
-        return (await this.beacon.getChainHead()).headSlot;
+    public onNewChainHead(callbacks: Array<any>, params: Array<any>): void {
```

Comment:
this should just accept single callback which you are suppose to persist in memory and call everytime there is new chainhead

Reply:
But that's what it does?

Reply:
it accepts array of <any> callbacks and params array.
You interface should be:
```javascript
public onNewChainHead(callback: (head: ChainHead) => void): void;
```

Reply:
Why? And not to support params? :(

Reply:
callback should bind it's own context to encourage best practices and testing